### PR TITLE
Bump Recurr and update CHANGELOG

### PR DESCRIPTION
**Pull request by romainallanot**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| Deprecations? | yes/no
| Tickets       | fixes #... 

This PR is an attempt to continue the work done in #65

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,28 +5,34 @@
...
+        "symfony/browser-kit": "^5.4 || ^6.0",
+        "symfony/css-selector": "^6.1",
+        "symfony/debug-bundle": "^5.4 || ^6.0",
+        "symfony/dotenv": "^5.4 || ^6.0",
+        "symfony/error-handler": "^6.1",
+        "symfony/intl": "^5.4 || ^6.0",
+        "symfony/panther": "^2.0",
+        "symfony/web-profiler-bundle": "^5.4 || ^6.0"
     },
+    "minimum-stability": "dev",
```
Comment (dunglas) [inline composer.json]: Is it mandatory?

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+services:
+  mysql:
+    image: mysql:8.0
+    environment:
+      MYSQL_ROOT_PASSWORD: mysql
+    ports:
+      - ${MYSQL_PORT:-3306}:3306
+
```
Comment (dunglas) [inline docker-compose.yaml]:
```suggestion
```

File: composer.json [hunk 3]
```diff
@@ -5,28 +5,34 @@
     "description": "Click and Collect plugin for Sylius, to sell and deliver securely during the COVID-19 pandemic.",
```
Comment (dunglas) [inline composer.json:5]: In a future PR, I would remove references to COVID-19, or maybe just in a "history" section in the README. It's a bit dated now.

File: src/Entity/ClickNCollectShipment.php [hunk 4]
```diff
@@ -21,20 +21,14 @@
  */
 trait ClickNCollectShipment
 {
-    /**
-     * @ORM\ManyToOne(targetEntity=\CoopTilleuls\SyliusClickNCollectPlugin\Entity\LocationInterface::class)
-     */
+    #[ORM\ManyToOne(targetEntity: LocationInterface::class)]
```
Comment (dunglas) [inline src/Entity/ClickNCollectShipment.php]: `targetEntity` and `type` properties should be optional when using attributes. The code can probably be simplified.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+## 0.3.0
```
Comment (dunglas) [inline CHANGELOG.md]: Maybe could we tag a 1.0? The code looks pretty stable now.

File: src/Validator/Constraints/SlotAvailable.php [hunk 6]
```diff
@@ -14,15 +14,13 @@
...
 
 use Symfony\Component\Validator\Constraint;
-use CoopTilleuls\SyliusClickNCollectPlugin\Validator\Constraints\SlotAvailableInterface;
 
 /**
  * Checks if a time slot is available.
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
- *
- * @Annotation
```
Comment (dunglas) [inline src/Validator/Constraints/SlotAvailable.php:24]: Maybe could we keep this for now to ease the migration from older versions?

File: tests/Application/.env [hunk 7]
```diff
@@ -12,12 +12,9 @@ APP_SECRET=EDITME
...
 ###< doctrine/doctrine-bundle ###
 
-###> symfony/swiftmailer-bundle ###
-# For Gmail as a transport, use: "gmail://username:password@localhost"
-# For a generic SMTP server, use: "smtp://localhost:25?encryption=&auth_mode="
-# Delivery is disabled by default via "null://localhost"
-MAILER_URL=smtp://localhost
-###< symfony/swiftmailer-bundle ###
+MAILER_DSN=nul://null
+
```
Comment (dunglas) [inline tests/Application/.env]:
```suggestion
```

**Review by dunglas — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (romainallanot) [reply composer.json]: No, I removed it

File: docker-compose.yaml (hunk 2 above)
Comment (romainallanot) [reply docker-compose.yaml]: Done

File: composer.json (hunk 3 above)
Comment (romainallanot) [reply composer.json:5]: In a future PR :)

File: src/Entity/ClickNCollectShipment.php (hunk 4 above)
Comment (romainallanot) [reply src/Entity/ClickNCollectShipment.php]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (romainallanot) [reply CHANGELOG.md]: OK done

File: src/Validator/Constraints/SlotAvailable.php (hunk 6 above)
Comment (romainallanot) [reply src/Validator/Constraints/SlotAvailable.php:24]: Done

File: tests/Application/.env (hunk 7 above)
Comment (romainallanot) [reply tests/Application/.env]: Done

File: src/Entity/ClickNCollectShipment.php (hunk 4 above)
Comment (dunglas) [reply src/Entity/ClickNCollectShipment.php]: You can remove similar props (such as `type` from all attributes.

**Pull request merged**